library/src/main/java/com/applandeo/materialcalendarview/utils/SelectedDay.kt

```diff
@@ -20,10 +20,14 @@ data class SelectedDay @JvmOverloads constructor(
 
     override fun equals(other: Any?): Boolean {
         return when (other) {
-            is SelectedDay -> calendar == other.calendar
-            is Calendar -> calendar == other
+            is SelectedDay -> calendar.compare(other.calendar)
+            is Calendar -> calendar.compare(other)
             else -> super.equals(other)
         }
     }
-}
 
+    private fun Calendar.compare(calendar: Calendar): Boolean =
+            this.get(Calendar.DAY_OF_MONTH) == calendar.get(Calendar.DAY_OF_MONTH) &&
```

Comment:
Shouldn't it be called something like `isTheSameDay` ?

Reply:
Changed ;)